File: src/mgx_api/schemas/agent.py
"""Agent/Task related schemas."""
from pydantic import BaseModel


class AgentTaskRequest(BaseModel):
    """Agent task request."""
    prompt: str


class AgentTaskResponse(BaseModel):
    """Agent task response."""
    task_id: str
    status: str


class AgentTaskResult(BaseModel):
    """Agent task result."""
    task_id: str
    status: str
    result: dict | None = None

File: frontend/src/types/agent.ts
/**
 * Agent related types
 */

export interface AgentTaskRequest {
  prompt: string;
}

export interface AgentTaskResponse {
  task_id: string;
  status: string;
}

export enum EventType {
  AGENT_START = 'agent_start',
  MESSAGE_DELTA = 'message_delta',
  MESSAGE_COMPLETE = 'message_complete',
  TOOL_START = 'tool_start',
  TOOL_END = 'tool_end',
  AGENT_ERROR = 'agent_error',
  FINISH = 'finish',
}

export interface SSEEvent {
  event_id: string;
  event_type: EventType;
  data: any;
}
